architecture/adr-list/adr-010-did-resolver.md

```diff
@@ -0,0 +1,89 @@
+
+# ADR 009: DID-resolver
+
+## Status
+
+| Category | Status |
+| :--- | :--- |
+| **Authors** | Renata Toktar|
+| **ADR Stage** | DRAFT |
+| **Implementation Status** | Not Implemented |
+| **Start Date** | 2022-02-22 |
+
+## Summary
+
+This document defines the architecture of two DID resolvers: Cheqd DID resolver and a universal DID resolver driver for integration with Universal resolver.
+
+## Context
+
+First, let's look at the discrepancies between the DID Doc model stored in the ledger and the format from the specification. Data is written to the ledger based on protobuffs, which is due to the use of the Cosmos framework. However, according to the specification, DID Doc must be provided in JSON format, which provides more features that are not achievable in protobuffs.
+Therefore, a new DID resolver is needed.
+
+## Decision
+
+Add a new web application for did-resolver
+
+Inconsistencies between DIDDoc from the ledger and specification that should be corrected:
+
+- Rename "context" to "@context" - will be done on the ledger side
+- Change snake_case to camelCase for field names - will be done on Cheqd resolver side using the marshaller setting
+- Remove empty lists  - will be done on Cheqd resolver side using the marshaller setting
+- Convert a list of pairs to a map for jwk_pubkey and other cases
+- In metadata make `did` property to be of `DID` type instead of `Any`
+- DID URL Dereferencing: handle links to provide DID fragments and convert them to the desired format - Cheqd DID resolver functionality.
+
+## Consequences
+
+### Cheqd resolver
+
+### Option 1 (chosen)
+
+Host the resolver separately from the ledger as an additional web service. Interaction with other applications and resolvers will implement [the following schema](https://drive.google.com/file/d/1pKL9I5fMhZ3TnAdkCiRTs53Y7zs9cGiv/view?usp=sharinghttps://drive.google.com/file/d/1pKL9I5fMhZ3TnAdkCiRTs53Y7zs9cGiv/view?usp=sharing):
```

Comment:
Is this the raw draw.io XML file? I'd recommend instead of a link to embed a PNG version and separately I'll export the draw.io file itself for editing and link